daemon/cdi_linux.go

```diff
@@ -0,0 +1,95 @@
+package daemon
+
+import (
+	"fmt"
+
+	"github.com/container-orchestrated-devices/container-device-interface/pkg/cdi"
+	"github.com/docker/docker/pkg/capabilities"
+	"github.com/hashicorp/go-multierror"
+	specs "github.com/opencontainers/runtime-spec/specs-go"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+// cdiDeviceInjector defines the operations required to inject CDI devices into an OCI spec.
+type cdiDeviceInjector interface {
+	InjectDevices(*specs.Spec, ...string) ([]string, error)
+	GetErrors() map[string][]error
+}
+
+type cdiHandler struct {
+	registry cdiDeviceInjector
+}
```

Comment:
```suggestion
type cdiHandler struct {
	registry *cdi.Cache
}
```
Note how the `errorOnUpdateSpec` change made the interface unnecessary.